apps/mobile/src/components/splash-screen-guard/splash-screen-guard.tsx

```diff
@@ -41,12 +42,25 @@ export function SplashScreenGuard({ children }: HasChildren) {
     void SplashScreen.hideAsync();
   }, [colorScheme, theme.colors]);
 
+  const checkUnlockTime = useCallback(() => {
+    return !!lastUnlocked && lastUnlocked > +new Date() - UNLOCK_TIMEOUT;
+  }, [lastUnlocked]);
+
   const tryAuthentication = useCallback(
     async ({ firstTry }: { firstTry: boolean }) => {
+      // if in insecure mode - just proceed with a splash
       if (securityLevelPreference !== 'secure') {
         playSplash();
         return;
       }
+      // if in secure mode, skip checks only if unlock time is not exceeding the timeout
+      // and this is not a cold start of the app
+      if (securityLevelPreference === 'secure' && authState !== 'cold-start' && checkUnlockTime()) {
+        setAuthState('passedOnFirst');
+        setAnimationFinished(true);
+        playSplash();
+        return;
+      }
```

Comment:
This timeout functionality seems kinda coupled specifically to the splash screen. Is there a way we could decouple better, perhaps with a hook that handles all timeout/auth concerns?